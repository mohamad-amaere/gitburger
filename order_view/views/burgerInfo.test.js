const path = require("path");
const ejs = require("ejs");
const { toMatchSnapshot } = require("jest-snapshot");

expect.extend({ toMatchSnapshot });

describe("burgerInfo.ejs", () => {
	it("should render correctly", async () => {
		const filePath = path.join(__dirname, "burgerInfo.ejs");
		const data = {
			burger: {
				burger_name: "Cheeseburger",
				price: 50,
				id: 1
			},
			ingredients: [
				{ id: 1, ingrediant_name: "Lettuce" },
				{ id: 2, ingrediant_name: "Tomato" },
				{ id: 3, ingrediant_name: "Cheese" }
			],
			cart: []
		};

		const html = await ejs.renderFile(filePath, data, { async: true });
		expect(html).toMatchSnapshot();
	});
});
