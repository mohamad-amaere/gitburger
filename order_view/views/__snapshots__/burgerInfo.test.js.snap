// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`burgerInfo.ejs should render correctly 1`] = `
"<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Gitburger - Cheeseburger</title>
  [object Promise]

</head>
<body>
  [object Promise]
  Cheeseburger
  <p>50 Kr</p>
  <form method="post" action="/burger/1">
    <input type="number" name="quantity" placeholder="quantity" min="1" max="5" value="1">
    
      <input type="checkbox" name="ingredients" value="1" checked>
      <label for="ingredient-1">Lettuce</label>
    
      <input type="checkbox" name="ingredients" value="2" checked>
      <label for="ingredient-2">Tomato</label>
    
      <input type="checkbox" name="ingredients" value="3" checked>
      <label for="ingredient-3">Cheese</label>
    
    <input type="text" value="50" name="price" hidden readonly>
    <button type="submit">best√§ll</button>
  </form>
</body>
</html>"
`;
